scheduler/src/main/java/io/camunda/zeebe/scheduler/ActorThread.java

```diff
@@ -97,21 +97,23 @@ private void doWork() {
 
   private void executeCurrentTask() {
     final var properties = currentTask.getActor().getContext();
-    MDC.setContextMap(properties);
+    boolean resubmit = false;
+
     MDC.put("actor-scheduler", actorThreadGroup.getSchedulerName());
+    for (final var property : properties.entrySet()) {
+      MDC.put(property.getKey(), property.getValue());
+    }
```

Comment:
❓ What is the difference over `#setContextMap`?

Reply:
`setContextMap` clears the previous map, so you can't have "static" members in there, i.e. values which you set once on thread creation and never change.